TalkAloud/TalkAloud/AudioEngine.swift

```diff
@@ -9,43 +9,52 @@
 import Foundation
 import AVFoundation
 
-// Struct will make it so I don't have to intialize variables
-// until I need to
-class AudioEngine {
-    var audioRecorder: AVAudioRecorder!
-    var audioPlayer: AVAudioPlayer!
-    var recordingSession: AVAudioSession!
-    var fileName = "selftalkfile.m4a"
+class AudioEngine: NSObject {
+    private var audioRecorder: AVAudioRecorder!
+    private var audioPlayer: AVAudioPlayer!
+    private var recordingSession: AVAudioSession!
+    private var fileName = "selftalkfile.m4a"
+    public private(set) var audioState: AudioEngineState = .stopped
+    private let audioRecordingSession = AVAudioSession.sharedInstance()
+    
+    public func getAudioRecordingSession() -> AVAudioSession {
```

Comment:
Remove this function.

Reply:
How else am I supposed to access the `audioRecordingSession` if I make the variable private in my class?

Reply:
Move the logic down below into the `record` function from the engine.
`try audioEngine.getAudioRecordingSession().setCategory(.playAndRecord, mode: .default)
try audioEngine.getAudioRecordingSession().setActive(true)`

Properties that are private should stay private. Exposing them trough functions like `getAudioRecordingSession` basically makes the private access control obsolete.

Reply:
Yeah, that was my solution a minute ago. 😄